File: src/main/java/mediathek/gui/dialogEinstellungen/allgemein/TextFieldConfigWriter.java
package mediathek.gui.dialogEinstellungen.allgemein;

import mediathek.tool.ApplicationConfiguration;

import javax.swing.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

/**
 * Generic config writer for textfields.
 * Action will be performed on Swing EDT.
 */
record TextFieldConfigWriter(JTextField control,
                             String configPropertyKey) implements ActionListener {

    @Override
    public void actionPerformed(ActionEvent e) {
        final String controlText = control.getText();
        ApplicationConfiguration.getConfiguration().setProperty(configPropertyKey, controlText);
    }
}

File: src/main/kotlin/mediathek/gui/dialogEinstellungen/allgemein/TextFieldConfigWriter.kt
package mediathek.gui.dialogEinstellungen.allgemein

import java.awt.event.ActionEvent
import java.awt.event.ActionListener
import java.util.function.Consumer
import javax.swing.JTextField

class TextFieldConfigWriter(
    private val control: JTextField,
    private val valueWriter: Consumer<String>,
) : ActionListener {
    override fun actionPerformed(e: ActionEvent) {
        valueWriter.accept(control.text)
    }
}
